components/http_foundation.rst

```diff
@@ -631,6 +631,105 @@ represented by a PHP callable instead of a string::
 
 .. _component-http-foundation-serving-files:
 
+Streaming a JSON Response
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 6.3
+
+    The :class:`Symfony\\Component\\HttpFoundation\\StreamedJsonResponse` class was
+    introduced in Symfony 6.3.
+
+The :class:`Symfony\\Component\\HttpFoundation\\StreamedJsonResponse` allows an API to
+return a lot of data as JSON and keep the used resources low by make usage of Generators.
+
+It expects an array which represents the JSON structure and the list which should be
+streamed are represented in the array as ``\Generator``. It also supports any kind of
+Traversable containing JSON Serializable data for a good developer experience,
+but for keep the resources usage low as possible it is recommended to use ``\Generators``,
+as they the advantages that only the current returned data need to be keep in memory.
+
+The response will stream the JSON with generators in to most efficient way and keep resources as low as possible::
+
+    use Symfony\Component\HttpFoundation\StreamedJsonResponse;
+
+    function loadArticles(): \Generator { // any method or function returning a Generator
+         yield ['title' => 'Article 1'];
+         yield ['title' => 'Article 2'];
+         yield ['title' => 'Article 3'];
+    };
+
+    $response = new StreamedJsonResponse(
+        // json structure with generators in which will be streamed as a list
+        [
+            '_embedded' => [
+                'articles' => loadArticles(), // any \Generator can be used which will be streamed as list of data
+            ],
+        ],
+    );
+
+.. tip::
+
+    If loading data via doctrine the ``toIterable`` method of ``Doctrine`` can be
+    used to keep also the resources low and fetch only row by row.
+    See the `Doctrine Batch processing`_ documentation for more::
+
+        public function __invoke(): Response
+        {
+            return new StreamedJsonResponse(
+                [
+                    '_embedded' => [
+                        'articles' => $this->loadArticles(),
+                    ],
+                ],
+            );
+        }
+
+        public function loadArticles(): \Generator
+        {
+            $queryBuilder = $entityManager->createQueryBuilder();
+            $queryBuilder->from(Article::class, 'article');
+            $queryBuilder->select('article.id')
+                ->addSelect('article.title')
+                ->addSelect('article.description');
+
+            return $queryBuilder->getQuery()->toIterable();
+        }
+
+.. tip::
+
+    If you have a lot of data to be returned you maybe want to call the
+    PHP `flush <https://www.php.net/manual/en/function.flush.php>`__ method between
```

Comment:
@javiereguiluz AFAIK we have a special notation for linking to native php function, right?